src/output/help.rs

```diff
@@ -666,13 +669,14 @@ impl<'b, 'c, 'd, 'w> Help<'b, 'c, 'd, 'w> {
             .collect::<Vec<_>>();
         let subcmds = self.parser.has_visible_subcommands();
 
-        let custom_headings = self.parser.app.args.args.iter().fold(0, |acc, arg| {
-            if arg.help_heading.is_some() {
-                acc + 1
-            } else {
-                acc
-            }
-        }) > 0;
+        let custom_headings = self
+            .parser
+            .app
+            .args
+            .args
+            .iter()
+            .filter_map(|arg| arg.help_heading)
+            .collect::<IndexSet<_>>();
```

Comment:
Why is this an `IndexSet` instead of `Vec`?

Reply:
We need to store headings in a container that will remember them just once and in the order in which they were inserted. `IndexSet` is a perfect fit for this, `Vec` is unsuitable here.

Reply:
Can you please explain a bit more? Is IndexSet faster than Vec? Also, even though we are currently doing the parsing only once, we might want to change it later for certain cases (repl, pre-parsing conditions etc..).

Reply:
Well, if I store into `Vec` it will just be a list of arguments' headings - with duplicates and out of order

For example, let's suppose we add the following arguments to the app:
```
arg1 - HeadingA
arg2 - HeadingB
arg3 - HeadingA
arg4 - HeadingB
arg5 - HeadingC
```

this will yield `[A, B, A, B, C]` for a `Vec`, but we need `[A, B, C]`. It's possible to do postprocessing to remove duplicates, but I expect this to be slower than just going for `IndexSet` anyway.

Reply:
Ah, I missed that. Thanks.